compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -509,6 +509,9 @@ impl<Tag: Copy, Extra> Allocation<Tag, Extra> {
             if Tag::ERR_ON_PARTIAL_PTR_OVERWRITE {
                 return Err(AllocError::PartialPointerOverwrite(first));
             }
+            warn!(
+                "Partial pointer overwrite! De-initializing memory at offsets {first:?}..{start:?}."
+            );
```

Comment:
I have `RUSTC_LOG=warn` set in my zshrc I believe. This will cause be rather noisy I think.

Reply:
They alsodon't get optimized out, so nightly will have them

Reply:
At least warn is only emitted when using `RUSTC_LOG=warn`.

Reply:
> I have RUSTC_LOG=warn set in my zshrc I believe. This will cause be rather noisy I think.

Note that this is never printed by rustc, since `ERR_ON_PARTIAL_PTR_OVERWRITE` is true there.
Only Miri can ever trigger this.

> They alsodon't get optimized out, so nightly will have them

This is a very cold path, so that's fine. We have `info!` for each statement and terminator we run (also exists on nightly), this is at least as important as that.

When this triggers, it can cause subtly wrong execution later in the program, where memory is uninit that should be init -- so there should be a good way to figure out of this triggered.

> We kind of emit warn and error unconditionally, or at least used to at some point.

This is definitely not printed by default, I verified that.